### PR TITLE
add software/pairix

### PR DESCRIPTION
**Pull request by jianhong**
## PR checklist

- [ x ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ x ] If necessary, include test data in your PR.
- [ x ] Remove all TODO statements.
- [ x ] Emit the `<SOFTWARE>.version.txt` file.
- [ x ] Follow the naming conventions.
- [ x ] Follow the parameters requirements.
- [ x ] Follow the input/output options guidelines.
- [ x ] Add a resource `label`
- [ x ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [ x ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ x ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

File: software/pairix/main.nf [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    path "*.version.txt"                      , emit: version
+
+    script:
+    def software = getSoftwareName(task.process)
+    def prefix   = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
+    """
+    pairix \\
+        -p pairs \\
+        $options.args \\
+        ${pair}
```
Comment (drpatelh) [inline software/pairix/main.nf:35]:
```suggestion
        $pair
```

File: software/pairix/main.nf [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    echo \$(samtools --help 2>&1) | sed 's/^.*Version: //; s/Usage.*\$//' > ${software}.version.txt
```
Comment (drpatelh) [inline software/pairix/main.nf:37]: Wrong version command here?

File: software/pairix/meta.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+      description: |
+        Groovy Map containing sample information
+        e.g. [ id:'test', single_end:false ]
+  - version:
+      type: file
+      description: File containing software version
+      pattern: "*.{version.txt}"
+  - pair:
+      type: file
+      description: pair file
```
Comment (drpatelh) [inline software/pairix/meta.yml:38]:
```suggestion
```

File: software/pairix/main.nf [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    input:
+    tuple val(meta), path(pair)
+
+    output:
+    tuple val(meta), path(pair), path("*.px2"), emit: index
+    path "*.version.txt"                      , emit: version
+
+    script:
+    def software = getSoftwareName(task.process)
+    def prefix   = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
```
Comment (drpatelh) [inline software/pairix/main.nf:30]:
```suggestion
```

Seems like this isn't used in the main `script` section so we can remove it.

**Review by jianhong — COMMENTED** [review]
Thank you Harshil. The module was update according the comments.

**Review by drpatelh — APPROVED** [review]
Nice 👌🏽

**Pull request merged**